src/wpool/mongoose_wpool_mgr.erl

```diff
@@ -70,7 +70,13 @@ start_link(Type) ->
 
 start(Type, Host, Tag, PoolOpts, ConnOpts) ->
     ok = ensure_started(Type),
-    gen_server:call(name(Type), {start_pool, Host, Tag, PoolOpts, ConnOpts}).
+    try
+        gen_server:call(name(Type), {start_pool, Host, Tag, PoolOpts, ConnOpts})
+    catch
+        _:_ ->
+            %% hide connection options
+            exit({"failed to start pool", Type, Host, Tag, PoolOpts})
```

Comment:
I'm sure we know how the password or any other secret field is actually called, so why don't we just log `maps:without(ConnOpts, [secret_1, secret_2...])`? Also, this approach looses stacktrace and converts possible `error` into `exit` (we wildcard the type of exception). 
One another thing to mention here is that using strings in error tuples makes them a bit useless, as it's quite hard to match on those if needed. 

That all being said, don't we have a better approach? How about passing secrets as functions, like "PasswordGetter" or something like that? It wouldn't come up in any logs, regardless what they are.

Reply:
actually we don't know. it's passed as provided in config file. and it's different for various backends.

Reply:
Actually it's hard to say where code can crash and whether some secrets are at this time somewhere in the call stack. With OTP reports, this means that the only way to make sure that secrets are not logged, ever, is to not pass them at all. Since this is not really possible, we can minimize this by e.g., like I mentioned, passing a functions instead. This makes the call stack that actually carries the secret limited to this one call that almost cannot crash and to whatever library that expects it does.

Reply:
we know that these changes are not perfect, they are addressing the exact reported issue. there will be some discussion regarding the further changes regarding this topic.

Reply:
> actually we don't know. it's passed as provided in config file. and it's different for various backends.

Then backends know that. I know that this is a bit out of scope, but I do think that each backend should define the configuration schema and as part of this definition, fields could be marked as "secret". 
As temporary solution, I wouldn't mind adding an optional callback to return list of possible secret fields. I do realize that this is more work, but current solution feels more like brute force solution employed in order to cover up the issue.

Reply:
that will not be implemented without preliminary discussion and so will not be implemented in this PR. there is no clean opinion on this topic within the team. We anyway cannot guaranty that external libraries are not exposing secrets on crashes, so it might be that we will just create some recommendation to avoid sharing logs externally.

Reply:
change rolled back, the further work in this area to be scheduled for the next release(s).